### PR TITLE
Changes to assert to ValueError in achor utils

### LINKED ISSUE #2947: Replace assert with ValueError in AnchorGenerator
## 🐛 Bug

I think with the Retina Net PR, there was an update to anchor utils.
My old code still works fine on torchvision 0.7 and pytorch 1.6. But on upgrading to 0.8 it doesn't.
I am not sure if this was BC Breaking change.

## To Reproduce

1. Following is the code block to create FRCNN
```
import torchvision
import torchvision.transforms as T
from torchvision.models.detection.rpn import AnchorGenerator
from torchvision.models.detection import FasterRCNN

device = torch.device('cuda') if torch.cuda.is_available() else torch.device('cpu')

def create_model(num_classes, min_size=300, max_size=500, backbone="mobile_net"):

    model = torchvision.models.detection.fasterrcnn_resnet50_fpn(pretrained=True)
    in_features = model.roi_heads.box_predictor.cls_score.in_features

    ft_mean = [0.485, 0.456, 0.406]
    ft_std = [0.229, 0.224, 0.225]

    mobile_net = torchvision.models.mobilenet_v2(pretrained=True)

    ft_backbone = mobile_net.features
    ft_backbone.out_channels = 1280

    ft_model = FasterRCNN(backbone=ft_backbone,
                        num_classes=num_classes, 
                        image_mean=ft_mean, 
                        image_std=ft_std,)
```
2. When I do These three function sequeantially
```

def load_model():
    detector = model.create_model(num_classes=config.NUM_CLASSES)
    detector.load_state_dict(torch.load(config.MODEL_SAVE_PATH, map_location=device))
    detector.eval()
    detector.to(device)
    return detector


def load_image_tensor(image_path, device):
    image_tensor = T.ToTensor() (Image.open(image_path))
    input_images = [image_tensor.to(device)]
    return input_images


def get_prediction(detector, images):
    with torch.no_grad():
        prediction = detector(images)
        return prediction

```

I get an assertion error from [this ](https://github.com/pytorch/vision/blob/master/torchvision/models/detection/anchor_utils.py#L103)line

## Expected behavior

It should work fine, I think an upgrade to torchvision should not break Anchor utils for FRCNN.

## Environment

 - PyTorch / torchvision Version (e.g., 1.0 / 0.4.0): PT -> 1.7  TV-> 0.8.1
 - OS (e.g., Linux): Windows
 - How you installed PyTorch / torchvision (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: NA
 - GPU models and configuration: NA

## Additional context

1. Maybe something has changed and I'm not aware of
2. Can we remove these assertions and raise `ValueError` with suitable messages ? I cannot debug at all with `assert` errors.
3. Slightly associated with #2955

### PR DESCRIPTION
**Pull request by oke-aditya**
Fixes #2947

Just a small change to ValueError instead of assert. This will enhance debugging.

### CODE REVIEW

File: torchvision/models/detection/anchor_utils.py [hunk 1]
```diff
@@ -100,7 +100,10 @@ def grid_anchors(self, grid_sizes, strides):
         anchors = []
         cell_anchors = self.cell_anchors
         assert cell_anchors is not None
-        assert len(grid_sizes) == len(strides) == len(cell_anchors)
+
+        if len(grid_sizes) != len(strides) != len(cell_anchors):
```
Comment (fmassa) [inline torchvision/models/detection/anchor_utils.py:104]: I don't think this equivalent to the previous implementation.
Quick sanity check:
```python
In [14]: 1 == 1 == 2
Out[14]: False

In [15]: 1 != 1 != 2
Out[15]: False
```
So the error messages will not be raised on some cases.

Can you do instead something like
```python
if not (... and .. and ...):
```

File: torchvision/models/detection/anchor_utils.py [hunk 2]
```diff
@@ -100,7 +100,10 @@ def grid_anchors(self, grid_sizes, strides):
...
+            raise ValueError('''Achors should be Tuple[Tuple[int]] because each feature
+                                map could potentially have different sizes and aspect ratios ''')
```
Comment (fmassa) [inline torchvision/models/detection/anchor_utils.py:106]: Can you use a single `"`, and not break the line as currently done, as it introduces artifacts on the print?
Also, can you make the error message a bit more descriptive, as you can have `Tuple[Tuple[int]]` being passed but still have the same problem. The underlying issue is that there needs to be a match between the number of feature maps passed and the number of sizes / aspect ratios specified.

**Review by fmassa — CHANGES_REQUESTED** [review]
Thanks for the PR!

I believe there is an issue with the logic, can you double-check?

File: torchvision/models/detection/anchor_utils.py (hunk 1 above)
Comment (oke-aditya) [reply torchvision/models/detection/anchor_utils.py:104]: Oh yes ! What a rookie mistake 😢  Let me fix that up

**Comment by oke-aditya** [comment]
Extremely sorry for such rookie error in code 🙏  
I have fixed it 👍 Also improved the error message.

**Review by fmassa — APPROVED** [review]
Thanks a lot!

For a follow-up PR, do you think you could send a PR adding a test for this ValueError?
Maybe in a file called `test_models_detection_anchor_utils.py`?

**Comment by oke-aditya** [comment]
Yep I will

**Pull request merged**